dashbuilder/dashbuilder-client/dashbuilder-navigation-client/src/main/java/org/dashbuilder/client/navigation/plugin/PerspectivePluginManagerImpl.java

```diff
@@ -0,0 +1,241 @@
+    private boolean pluginsLoaded = false;
+    private Stack<NavGroup> navGroupStack = new Stack<>();
+
+    @Inject
+    public PerspectivePluginManagerImpl(ClientTypeRegistry clientTypeRegistry,
+                                    LayoutGenerator layoutGenerator,
+                                    NavigationManager navigationManager,
+                                    Caller<PerspectivePluginServices> pluginServices,
+                                    Event<PerspectivePluginsChangedEvent> perspectivesChangedEvent) {
+        this.clientTypeRegistry = clientTypeRegistry;
+        this.layoutGenerator = layoutGenerator;
+        this.navigationManager = navigationManager;
+        this.pluginServices = pluginServices;
+        this.perspectivesChangedEvent = perspectivesChangedEvent;
+    }
+
+    @Override
+    public void getPerspectivePlugins(ParameterizedCommand<Collection<Plugin>> callback) {
+        loadPlugins(callback);
+    }
+
+
+    @Override
+    public boolean isRuntimePerspective(Plugin plugin) {
+        ClientResourceType type = clientTypeRegistry.resolve(plugin.getPath());
+        return type != null && type instanceof PerspectiveLayoutPluginResourceType;
+    }
+
+    @Override
+    public boolean isRuntimePerspective(NavItem navItem) {
+        return getRuntimePerspectiveId(navItem) != null;
+    }
+
+    @Override
+    public boolean isRuntimePerspective(String perspectiveId) {
+        return pluginMap.containsKey(perspectiveId);
+    }
+
+    @Override
+    public String getRuntimePerspectiveId(NavItem navItem) {
+        NavWorkbenchCtx navCtx = NavWorkbenchCtx.get(navItem);
+        String resourceId = navCtx.getResourceId();
+        ActivityResourceType resourceType = navCtx.getResourceType();
+        boolean isRuntimePerspective = resourceId != null && ActivityResourceType.PERSPECTIVE.equals(resourceType) && isRuntimePerspective(resourceId);
+        return isRuntimePerspective ? resourceId : null;
+    }
+
+    @Override
+    public boolean existsPerspectivePlugin(String perspectiveName) {
+        return perspectiveName != null && pluginMap.get(perspectiveName) != null;
+    }
+
+    @Override
+    public void getLayoutTemplateInfo(String perspectiveName, ParameterizedCommand<LayoutTemplateInfo> callback) {
+        pluginServices.call((RemoteCallback<LayoutTemplateInfo>) callback::execute).getLayoutTemplateInfo(perspectiveName);
+    }
+
+    @Override
+    public void getLayoutTemplateInfo(LayoutTemplate layoutTemplate, ParameterizedCommand<LayoutTemplateInfo> callback) {
+        pluginServices.call((RemoteCallback<LayoutTemplateInfo>) callback::execute).getLayoutTemplateInfo(layoutTemplate);
+    }
+
+    @Override
+    public void buildPerspectiveWidget(String perspectiveName, LayoutTemplateContext layoutCtx, ParameterizedCommand<IsWidget> afterBuild, ParameterizedCommand<LayoutRecursionIssue> onInfiniteRecursion) {
+        Plugin plugin = pluginMap.get(perspectiveName);
```

Comment:
I have to agree but this code was a refactor from PerspectivePluginManager. We had to re-implement the behavior of this class in Runtime, so I had to extract and interface out of it and add a specific implementation in Runtime! Hence why I didn't modify any of the original code to avoid breaking stuff in other webapps


[1] https://github.com/kiegroup/appformer/blob/master/dashbuilder/dashbuilder-client/dashbuilder-navigation-client/src/main/java/org/dashbuilder/client/navigation/plugin/PerspectivePluginManager.java#L147